sstables/storage.cc

```diff
@@ -71,6 +71,9 @@ class filesystem_storage final : public sstables::storage {
     }
 
     virtual sstring prefix() const override { return dir; }
+    virtual sstring location(const sstable& sst, std::optional<component_type> f = std::nullopt) const override {
```

Comment:
if we default to the "data" component, why not just pass `component_type` here and use `component_type::Data` as the default value?

Reply:
S3 .location() tells nullopt from Data